### PR TITLE
Add serverless DW platform

### LINKED ISSUE #24115: Add SQL Projects support for Serverless DW
Add Serverless DW as a target platform for database projects

### PR DESCRIPTION
**Pull request by zijchen**
Fixes #24115 
Add SqlServerless target platform
Extract project from serverless DW
Tested locally

### CODE REVIEW

**Review by kisantia — APPROVED** [review]
Can you also please verify that the master system dacpac package reference is working as expected for synapse serverless? It should be handled by the current version of Microsoft.SqlServer.DacFx.Projects in STS, but would be good to verify now that this target platform is available.

File: extensions/sql-database-projects/src/common/utils.ts [hunk 1]
```diff
@@ -731,8 +731,8 @@ export async function getTargetPlatformFromServerVersion(serverInfo: azdataType.
 		if (azdataApi) {
 			// TODO: Update this when Fabric DW gets its own engine edition
```
Comment (Benjin) [inline extensions/sql-database-projects/src/common/utils.ts:732]: Is this change the update mentioned in this TODO?

File: extensions/sql-database-projects/src/common/constants.ts [hunk 2]
```diff
@@ -629,6 +629,7 @@ export const targetPlatformToVersion: Map<string, string> = new Map<string, stri
 	[SqlTargetPlatform.sqlServer2022, '160'],
 	[SqlTargetPlatform.sqlAzure, 'AzureV12'],
 	[SqlTargetPlatform.sqlDW, 'Dw'],
+	[SqlTargetPlatform.sqlDwServerless, 'Serverless'],
```
Comment (Benjin) [inline extensions/sql-database-projects/src/common/constants.ts:632]: This maps to Database Schema Provider in the .sqlproj, correct?  If so, can you update the test that validates support? https://msdata.visualstudio.com/SQLToolsAndLibraries/_git/SqlProjects?path=/Test/ProjectTests.cs&version=GBmain&line=201&lineEnd=201&lineStartColumn=27&lineEndColumn=52&lineStyle=plain&_a=contents

**Review by Benjin — APPROVED** [review]

**Comment by zijchen** [comment]
> Can you also please verify that the master system dacpac package reference is working as expected for synapse serverless? It should be handled by the current version of Microsoft.SqlServer.DacFx.Projects in STS, but would be good to verify now that this target platform is available.

Unfortunately it looks like it's broken, the reference is added like this:
```xml
    <PackageReference Include="Microsoft.SqlServer.Dacpacs.Master">
      <Version>Serverless.0.0</Version>
      <GeneratePathProperty>True</GeneratePathProperty>
      <DatabaseVariableLiteralValue>master</DatabaseVariableLiteralValue>
    </PackageReference>
```

I added the master.dacpac directly for Serverless to ArtifactReference works now. I'll look in SqlProj code to see where I need to update

**Comment by kisantia** [comment]
> > Can you also please verify that the master system dacpac package reference is working as expected for synapse serverless? It should be handled by the current version of Microsoft.SqlServer.DacFx.Projects in STS, but would be good to verify now that this target platform is available.
> 
> Unfortunately it looks like it's broken, the reference is added like this:
> 
> ```
>     <PackageReference Include="Microsoft.SqlServer.Dacpacs.Master">
>       <Version>Serverless.0.0</Version>
>       <GeneratePathProperty>True</GeneratePathProperty>
>       <DatabaseVariableLiteralValue>master</DatabaseVariableLiteralValue>
>     </PackageReference>
> ```
> 
> I added the master.dacpac directly for Serverless to ArtifactReference works now. I'll look in SqlProj code to see where I need to update

I'm guessing it broke when the SqlPlatformNames was updated since the sql projects code is looking for the platform name `SqlServerless`, so hopefully just a small update needed to get it working again.

File: extensions/sql-database-projects/src/common/constants.ts (hunk 2 above)
Comment (zijchen) [reply extensions/sql-database-projects/src/common/constants.ts:632]: Added https://msdata.visualstudio.com/SQLToolsAndLibraries/_git/SqlProjects/pullrequest/1147824

**Pull request merged**